src/main/java/org/zalando/nakadi/service/converter/VersionOneConverter.java

```diff
@@ -0,0 +1,72 @@
+package org.zalando.nakadi.service.converter;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.zalando.nakadi.domain.CursorError;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.NakadiCursor;
+import org.zalando.nakadi.domain.Timeline;
+import org.zalando.nakadi.exceptions.InternalNakadiException;
+import org.zalando.nakadi.exceptions.InvalidCursorException;
+import org.zalando.nakadi.exceptions.NoSuchEventTypeException;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.service.timeline.TimelineService;
+import org.zalando.nakadi.view.Cursor;
+
+class VersionOneConverter implements VersionedConverter {
+    private static final int TIMELINE_ORDER_LENGTH = 4;
+    private static final int TIMELINE_ORDER_BASE = 16;
+
+    private final EventTypeCache eventTypeCache;
+    private final TimelineService timelineService;
+
+    VersionOneConverter(final EventTypeCache eventTypeCache, final TimelineService timelineService) {
+        this.eventTypeCache = eventTypeCache;
+        this.timelineService = timelineService;
+    }
+
+    @Override
+    public NakadiCursor.Version getVersion() {
+        return NakadiCursor.Version.ONE;
+    }
+
+    public NakadiCursor convert(final String eventTypeStr, final Cursor cursor)
+            throws InternalNakadiException, NoSuchEventTypeException, InvalidCursorException {
+        final String orderStr = cursor.getOffset().substring(
+                NakadiCursor.VERSION_LENGTH, NakadiCursor.VERSION_LENGTH + TIMELINE_ORDER_LENGTH);
+        final int order = Integer.parseInt(orderStr, TIMELINE_ORDER_BASE);
+        final List<Timeline> timelines = eventTypeCache.getTimelines(eventTypeStr);
+        final EventType eventType = eventTypeCache.getEventType(eventTypeStr);
+        final String realOffset = cursor.getOffset().substring(NakadiCursor.VERSION_LENGTH + TIMELINE_ORDER_LENGTH);
```

Comment:
I think that will make sense to catch StringIndexOutOfBoundsException and throw InvalidCursorException here